client/state/posts/selectors.js

```diff
@@ -34,23 +34,32 @@ export function getPost( state, globalId ) {
 }
 
 /**
- * Returns a normalized post object by its global ID.
+ * Returns a normalized post object by its global ID, or null if the post does
+ * not exist. A normalized post includes common transformations to prepare the
+ * post for display.
  *
- * @param  {Object} state    Global state tree
- * @param  {String} globalId Post global ID
- * @return {Object}          Post object
+ * @param  {Object}  state    Global state tree
+ * @param  {String}  globalId Post global ID
+ * @return {?Object}          Post object
  */
 export const getNormalizedPost = createSelector(
 	( () => {
-		// Cache normalize flow in immediately-invoked closure so to avoid
+		// Cache normalize flow in immediately-invoked closure to avoid
 		// regenerating same flow on each call to this selector
 		const normalize = flow( [
 			firstPassCanonicalImage,
 			decodeEntities,
 			stripHtml
 		] );
 
-		return ( state, globalId ) => normalize( cloneDeep( getPost( state, globalId ) ) );
+		return ( state, globalId ) => {
+			const post = getPost( state, globalId );
+			if ( ! post ) {
+				return null;
+			}
+
+			return normalize( cloneDeep( post ) );
```

Comment:
The problem with storing normalized posts as the "canonical" post in the state tree is that it might cause some issues with editing posts (where the REST API won't expect these fields if they were sent). I personally feel normalization is a good use-case for state selectors as we've done here. I might even suggest to go one step further and apply individual normalization rules as separate selectors, but that's probably a step backwards in terms of simplifying the experience for developers (the "bad" experience being as in previously where developer had to manually call `decodeEntities` on titles prior to this selector existing, which was easily overlooked).